### PR TITLE
Policy Output Improvements

### LINKED ISSUE #287: Misalignment of Policy Output
### Describe the Bug

When name of policy or description is long columns do not align.

Screen Shot 2021-11-17 at 5 20 32 PM

### Expected Behavior

Data/output should be well aligned

### PR DESCRIPTION
**Pull request by bbernays**
Fixes #287

### CODE REVIEW

File: pkg/ui/console/client.go [hunk 1]
```diff
@@ -540,14 +540,37 @@ func printPolicyResponse(results []*policy.ExecutionResult) {
...
+		for _, res := range execResult.Results {
+			if len(res.Name) > maxNameLength {
+				maxNameLength = len(res.Name) + 1
+			}
+			if len(res.Description) > maxDescrLength {
+				maxDescrLength = len(res.Description) + 1
+			}
+		}
+
+		fmtString := fmt.Sprintf("\t%%s  %%-%ds %%-%ds %%%ds\n", maxNameLength, maxDescrLength, 10)
```
Comment (roneli) [inline pkg/ui/console/client.go:554]: maybe move this to a separate function? add a small unitest to it?

File: pkg/ui/console/client.go [hunk 2]
```diff
@@ -540,14 +540,37 @@ func printPolicyResponse(results []*policy.ExecutionResult) {
...
 			case res.Type == policy.ManualQuery:
-				ui.ColorizedOutput(ui.ColorInfo, "\t%s  %-10s %-120s %10s\n", emojiStatus[ui.StatusWarn], res.Name, res.Description, color.YellowString("manual"))
+				ui.ColorizedOutput(ui.ColorInfo, fmtString, emojiStatus[ui.StatusWarn], res.Name, res.Description, color.YellowString("manual"))
 			default:
-				ui.ColorizedOutput(ui.ColorInfo, "\t%s %-10s %-120s %10s\n", emojiStatus[ui.StatusError], res.Name, res.Description, color.RedString("failed"))
+				ui.ColorizedOutput(ui.ColorInfo, fmtString, emojiStatus[ui.StatusError], res.Name, res.Description, color.RedString("failed"))
+				ui.ColorizedOutput(ui.ColorWarning, "\n")
+				// specific columns can be decided upon later
+				for index, column := range res.Columns {
+					if column == "cq_output" || column == "arn" || column == "id" {
```
Comment (roneli) [inline pkg/ui/console/client.go:567]: would makes this in a variable i.e var CommonIdentifierColumns = []string{}, and check if column contained and print.

File: pkg/ui/console/client.go [hunk 3]
```diff
@@ -540,14 +540,37 @@ func printPolicyResponse(results []*policy.ExecutionResult) {
...
+						for _, row := range res.Data {
+							ui.ColorizedOutput(ui.ColorInfo, "\t\t%s  %-10s \n", emojiStatus[ui.StatusError], fmt.Sprintf("%v", row[index]))
+							ui.ColorizedOutput(ui.ColorWarning, "\n")
```
Comment (roneli) [inline pkg/ui/console/client.go:570]: would move this print to its own function i.e printOffendingResources ? maybe a better name

**Review by roneli — CHANGES_REQUESTED** [review]
Looks good, minor nit picks

File: pkg/ui/console/client.go (hunk 2 above)
Comment (bbernays) [reply pkg/ui/console/client.go]: I thought about doing that, but cannot just check for "contains" as their is a precedence

File: pkg/ui/console/client.go [hunk 4]
```diff
@@ -539,21 +541,93 @@ func printPolicyResponse(results []*policy.ExecutionResult) {
...
+		if outputKey != "" {
+			break
+		}
+	}
+	if outputKey != "" {
+		for index, column := range columnNames {
+			if column == outputKey {
+				for _, row := range data {
+					return fmt.Sprintf("%v", row[index])
+				}
```
Comment (roneli) [inline pkg/ui/console/client.go:624]: invert this if

File: pkg/ui/console/client.go [hunk 5]
```diff
@@ -539,21 +541,93 @@ func printPolicyResponse(results []*policy.ExecutionResult) {
...
+		for _, column := range columnNames {
+			if key == column {
+				outputKey = key
+			}
+		}
+		if outputKey != "" {
+			break
+		}
+	}
+	if outputKey != "" {
```
Comment (roneli) [inline pkg/ui/console/client.go:619]: invert this if

File: pkg/ui/console/client.go [hunk 6]
```diff
@@ -539,21 +541,93 @@ func printPolicyResponse(results []*policy.ExecutionResult) {
...
+		if len(res.Name) > maxNameLength {
+			maxNameLength = len(res.Name) + 1
+		}
+		if len(res.Description) > maxDescrLength {
+			maxDescrLength = len(res.Description) + 1
+		}
+	}
+
+	return fmt.Sprintf("\t%%s  %%-%ds %%-%ds %%%ds\n", maxNameLength, maxDescrLength, 10)
+
```
Comment (roneli) [inline pkg/ui/console/client.go:603]: nit: remove newline

**Review by roneli — APPROVED** [review]
LGTM, minor nit + suggestions

File: pkg/ui/console/client.go (hunk 6 above)
Comment (bbernays) [reply pkg/ui/console/client.go:608]: Done

File: pkg/ui/console/client.go (hunk 5 above)
Comment (bbernays) [reply pkg/ui/console/client.go]: Done

File: pkg/ui/console/client.go (hunk 4 above)
Comment (bbernays) [reply pkg/ui/console/client.go]: Done

**Review by roneli — APPROVED** [review]
LGTM

**Pull request merged**